cosmwasm/contract.ts

```diff
@@ -0,0 +1,433 @@
+import { StdFee } from '@cosmjs/stargate';
+import { Command, Option } from 'commander';
+
+import { getChainConfig, itsEdgeContract, printInfo, prompt } from '../common';
+import { ConfigManager, GATEWAY_CONTRACT_NAME, VERIFIER_CONTRACT_NAME } from '../common/config';
+import { addAmplifierOptions } from './cli-utils';
+import { CoordinatorManager } from './coordinator';
+import { ClientManager, Options } from './processor';
+import { mainProcessor } from './processor';
+import { execute } from './submit-proposal';
+import { executeTransaction, getChainTruncationParams, getCodeId, usesGovernanceBypass, validateItsChainChange } from './utils';
+
+interface ContractCommandOptions extends Omit<Options, 'contractName'> {
+    yes?: boolean;
+    title?: string;
+    description?: string;
+    chains?: string[];
+    itsEdgeContract?: string;
+    itsMsgTranslator?: string;
+    update?: boolean;
+    contractName?: string;
+    msg?: string[];
+    epochDuration?: string;
+    participationThreshold?: string;
+    rewardsPerEpoch?: string;
+    salt?: string;
+    admin?: string;
+    gatewayCodeId?: number;
+    verifierCodeId?: number;
+    proverCodeId?: number;
+    fetchCodeId?: boolean;
+    [key: string]: unknown;
+}
+
+const confirmDirectExecution = (options: ContractCommandOptions, messages: string[], contractAddress: string): boolean => {
+    printInfo('Contract address', contractAddress);
+
+    messages.forEach((msg, index) => {
+        const message = typeof msg === 'string' ? JSON.parse(msg) : msg;
+        printInfo(`Message ${index + 1}/${messages.length}`, JSON.stringify(message, null, 2));
+    });
+
+    if (prompt('Proceed with direct execution?', options.yes)) {
```

Comment:
prompts are used for either proceeding or cancelling. if no corresponds to a choice, this doesn't work in github actions where you can't respond to prompts